eng/pipelines/libraries/helix-queues-setup.yml

```
@@ -150,8 +153,10 @@ jobs:
           - ${{ if or(eq(parameters.jobParameters.isExtraPlatformsBuild, true), eq(parameters.jobParameters.includeAllPlatforms, true)) }}:
             - Windows.11.Amd64.Client.Open
             - Windows.Amd64.Server2022.Open
+            - Windows.Server2025.Amd64.Open
           - ${{ if or(ne(parameters.jobParameters.isExtraPlatformsBuild, true), eq(parameters.jobParameters.includeAllPlatforms, true)) }}:
             - Windows.Amd64.Server2022.Open
+            - Windows.Server2025.Amd64.Open
```

Existing, but the logic here seems suspect.  It says

```
if (extraPlatforms || allPlatforms)
 - A, B, C

if (!extraplatforms || allPlatforms)
 - B, C
```

So effectively both B,C are always set so maybe we could pull them out of the conditions.
```suggestion
          - Windows.Amd64.Server2022.Open
          - Windows.Server2025.Amd64.Open
          - ${{ if or(eq(parameters.jobParameters.isExtraPlatformsBuild, true), eq(parameters.jobParameters.includeAllPlatforms, true)) }}:
            - Windows.11.Amd64.Client.Open
```

Which is interesting - seems like we are treating Server as the primary test platform.

---

Thanks - I didn't trust my eyes on this one - I'm not YAML native, so it was curious why the logic looked like that.

---

It probably grew organically - perhaps at one time there were larger differences between these?  Maybe folks were trying to have conditions that resolved to complete sets.  Hard to tell.  Now it seems to be somewhat of a mess.